### DISCUSSION TITLE
Workflow-core capabilities and my project requirements

### DISCUSSION THREAD

**Discussion by devtekk** [danielgerlag/workflow-core#951] [General]
Hi,

I am working on converting "paper form" processes to online (intranet site). Interested in using Workflow-core but had some questions to see what it does "as is"  and what could be modified.

The processes will have multiple people, sometimes in role groups or sometimes specifically assigned, to complete action steps. The processes would not be linear as someone could return the request to an earlier action to change existing data. Also an action could be saved as a "draft" for the person to later return to it.

Example of actions:

1) employee submits request.
2) employee supervisor reviews and approves, denies or sends back to employee for a change.
3) Budget staff review request (entering data). This action is open to any of the users designated as Budget staff. A dashboard would show a list of open requests at that step and they click a link to open a request automatically assigning themselves. Budget could return request back to a previous step.
4) Budget Director (one pre-assigned person) approves. Budget Director could return the request to Budget Staff to change something.

5) If needed and selected, Manager's supervisor (assigned in Budget staff action) approves or sends to back to a previous step. 
6) If needed and selected,  Director would review and approve/decline/require change.
7) Purchasing staff complete an action much like Budget Staff situation in step 2 above.

Also need to include a dashboard to show open requests if user is in a group, show history of a request, search, etc.

Only the users involved in a request could see the info. HR and Purchasing staff would be able to view a request at any point and also reassign themselves if a group member had started but not completed an action.

Thank you.

**Comment by neozhu** [comment]
my opinion is that there is no need to use workflow-core. Many features of the manual approval workflow are not provided, such as flow designer, history,  task Kanban, and personal organization
I think this workflow-core is better suited for developing a group of automated activities make a process flow
